lib/semantics/check-do-concurrent.cc

```diff
@@ -401,7 +401,6 @@ class DoConcurrentContext {
     : messages_{context.messages()} {}
 
   bool operator==(const DoConcurrentContext &x) const { return this == &x; }
-  bool operator!=(const DoConcurrentContext &x) const { return this != &x; }
```

Comment:
Why have `operator==` without `operator!=`?

Reply:
I realized that classes that were used as part of `Indirection<>` had to define unused `operator!=` because `Indirection<>` itself has an unused `operator!=`.  So I cleaned them up.